homeassistant/components/lgthinq/device.py

```diff
@@ -0,0 +1,439 @@
+"""Implements LG ThinQ device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from thinqconnect import PROPERTY_READABLE, DeviceType, ThinQApi
+from thinqconnect.devices.connect_device import ConnectBaseDevice, ConnectDeviceProfile
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEVICE_TYPE_API_MAP, DOMAIN, NONE_KEY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LGDevice:
```

Comment:
Okay so what exactly is the role of the LGDevice in the integration?

Reply:
1. We are planning to release it in October.
2. A LGDevice represents a single LG appliance device which is different from typical IoT devices.
We need to handle devices' data, notification and status through LG Device instance.
It's because of ThinQ Cloud structures which is also communicating with mobile apps.
It also works as a coordinator for a device.

Reply:
> A LGDevice represents a single LG appliance device which is different from typical IoT devices

I don't really see a difference in how it's different from typical IoT devices and would love to get to know more about this so I can get in the same flow of decision making. I think using the common practices we have in Home Assistant is more maintainable in the long run.

Is it possible to maybe chat on Discord with one of you?